pkg/notify/notifier/feishu/feishu.go

```diff
@@ -204,106 +216,175 @@ func (n *Notifier) sendToChatBot(ctx context.Context, message *Message) error {
 		}
 	}
 
-	if err := n.addKeywords(message); err != nil {
-		_ = level.Error(n.logger).Log("msg", "FeishuNotifier: add keywords error", "error", err.Error())
-		return err
-	}
+	send := func() (bool, error) {
+		webhook, err := n.notifierCtl.GetCredential(n.receiver.ChatBot.Webhook)
+		if err != nil {
+			return false, err
+		}
 
-	var buf bytes.Buffer
-	if err := utils.JsonEncode(&buf, message); err != nil {
-		_ = level.Error(n.logger).Log("msg", "FeishuNotifier: encode message error", "error", err.Error())
-		return err
-	}
+		var buf bytes.Buffer
+		if err := utils.JsonEncode(&buf, message); err != nil {
+			return false, err
+		}
 
-	request, err := http.NewRequest(http.MethodPost, webhook, &buf)
-	if err != nil {
-		return err
-	}
-	request.Header.Set("Content-Type", "application/json; charset=utf-8")
+		request, err := http.NewRequest(http.MethodPost, webhook, &buf)
+		if err != nil {
+			return false, err
+		}
+		request.Header.Set("Content-Type", "application/json; charset=utf-8")
 
-	respBody, err := utils.DoHttpRequest(ctx, nil, request)
-	if err != nil {
-		_ = level.Error(n.logger).Log("msg", "FeishuNotifier: do http error", "error", err)
-		return err
-	}
+		respBody, err := utils.DoHttpRequest(ctx, nil, request)
+		if err != nil && len(respBody) == 0 {
+			return false, err
+		}
 
-	var resp Response
-	if err := utils.JsonUnmarshal(respBody, &resp); err != nil {
-		_ = level.Error(n.logger).Log("msg", "FeishuNotifier: decode response body error", "error", err)
-		return err
-	}
+		var resp Response
+		if err := utils.JsonUnmarshal(respBody, &resp); err != nil {
+			return false, err
+		}
 
-	if resp.Code != 0 {
-		_ = level.Error(n.logger).Log("msg", "FeishuNotifier: send message to chatbot error", "code", resp.Code, "message", resp.Msg)
-		return utils.Errorf("%d, %s", resp.Code, resp.Msg)
-	}
+		if resp.Code == 0 {
+			return false, nil
+		}
 
-	return nil
-}
+		if resp.Code == 9499 {
```

Comment:
9499 is feishu's code indicating need to retry?
Maybe we should define a const for that which is meaningful , or add a comment here